### PR TITLE
Suggestion view

### PR DESCRIPTION
**Pull request by dcarrilloWolox**
## Summary ##
* I create a new book suggestion view
## Screenshots ##
screencapture-localhost-3000-suggestions-new-1502895557863

## Trello card ##
https://trello.com/c/7mUhpXYf/28-crear-vista-para-crear-book-suggestions

### CODE REVIEW

File: app/controllers/book_suggestions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class BookSuggestionsController < ApplicationController
+  def new
+    @users = User.all
+    @suggestion = BookSuggestion.new
+  end
+
+  def create
+    suggestion_params = params.require(:book_suggestion).permit!
```
Comment (Martum) [inline app/controllers/book_suggestions_controller.rb]: Its not a good practice use `.permit!` lets change it with strong params!

File: app/controllers/book_suggestions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class BookSuggestionsController < ApplicationController
+  def new
+    @users = User.all
+    @suggestion = BookSuggestion.new
+  end
+
+  def create
```
Comment (Martum) [inline app/controllers/book_suggestions_controller.rb]: You are POSTing to this action via AJAX, move it to Api/V1

File: app/views/book_suggestions/new.html.slim [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    th
+      | Year
+    th
+      | Editorial
+    th
+      | Price
+javascript:
+  $(document).ready(function(){
+    $('#suggestion-form')
+    .on('ajax:success',
```
Comment (Martum) [inline app/views/book_suggestions/new.html.slim]: Add error handling! Nothing too fancy, just an error box displaying all the errors.

File: app/controllers/book_suggestions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    suggestion_params = params.require(:book_suggestion).permit!
+    suggestion = BookSuggestion.new(suggestion_params)
+
+    if suggestion.save
+      head :created
+    else
+      render suggestion.errors
```
Comment (Martum) [inline app/controllers/book_suggestions_controller.rb]: `render json: suggestion.errors.to_h`

**Review by Martum — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/new_book_suggestion.scss [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+  input {
+    background-color: $orange;
+    border-radius: 5px;
+    height: 40px;
+    width: 200px;
+  }
+}
+
+#suggestion-table {
```
Comment (JessSaavedra) [inline app/assets/stylesheets/new_book_suggestion.scss]: Don't use ids in CSS (even though it's ok for grabbing an element in javascript). They have higher priority than simpler rules and that can cause trouble when adding new style to a component.

File: app/views/book_suggestions/new.html.slim [hunk 6]
```diff
@@ -0,0 +1,71 @@
+h1
+  | New Suggestion
+= form_with model: @suggestion, html: {class: ['flex-column', 'new-form'], id: 'suggestion-form'} do |f|
```
Comment (JessSaavedra) [inline app/views/book_suggestions/new.html.slim]: You can directly write `class: 'flex-column new-form'`

File: app/views/book_suggestions/new.html.slim [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+      | Publisher
+    th
+      | Link
+    th
+      | Year
+    th
+      | Editorial
+    th
+      | Price
+javascript:
```
Comment (JessSaavedra) [inline app/views/book_suggestions/new.html.slim]: It's not recommended to mix rails code with javascript, let's move this to a separated file in app/assets/javascripts

File: app/views/book_suggestions/new.html.slim [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  $(document).ready(function(){
+    $('#suggestion-form')
+    .on('ajax:success',
+      function(e, data, status, xhr){
+        console.log($('form').serializeArray().length);
```
Comment (JessSaavedra) [inline app/views/book_suggestions/new.html.slim]: Delete this console.log

File: app/views/book_suggestions/new.html.slim [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        let formData = $('form').serializeArray().slice(2,9);
+        
+        addSuggestion(formData);
+        showTable();
+      });
+      
+    var showTable = function() {
+      $('#suggestion-table').css("display", "block");
```
Comment (JessSaavedra) [inline app/views/book_suggestions/new.html.slim]: With jquery, you can use `$('#suggestion-table').toggle()` which alternates between showing and hiding the element. Good thing about it is that you can use it for both hiding and showing and that you are not coupled to knowing which is the exact display of the element before hiding it (unless you actually want to change it to be differente from original CSS, but I think it's not a common case :stuck_out_tongue: )

File: app/views/book_suggestions/new.html.slim [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+    };
+    
+    var addSuggestion = function(suggestions) {
```
Comment (JessSaavedra) [inline app/views/book_suggestions/new.html.slim]: I think it's confusing to have this parameter called suggestions, wouldn't it be just a suggestion? Maybe you can name this suggestionFields or something like that? :thinking:

**Review by JessSaavedra — CHANGES_REQUESTED** [review]

File: app/controllers/book_suggestions_controller.rb (hunk 2 above)
Comment (Martum) [reply app/controllers/book_suggestions_controller.rb]: `skip_before_action :current_user, :authenticate`

**Pull request merged**